changes/1217.bugfix.rst

```diff
@@ -0,0 +1 @@
+macOS apps are now guaranteed to be universal binaries when dependencies only provide single-architecture binary wheels.
```

Comment:
Clarify that the intention is for ALL apps to be universal binaries.

```suggestion
macOS apps are now guaranteed to be universal binaries, even when dependencies only provide single-architecture binary wheels.
```

Reply:
What's the path forward when packages stop releasing x86_64 and universal2 wheels, and focus only on arm64? I guess we'll need to re-introduce an option to build single-platform apps again.

Reply:
I thought about that - it's potentially even an issue right now, if you have a particularly progressive app developer who decides that they either (a) don't care about x86 users, or (b) are impacted by an upstream dependency that doesn't provide an x86 package. Both are entirely possible outcomes - given that there are projects already deprecated Python 3.8 support, I wouldn't think we're *too* far off this being a practical concern.

My inclination is to land this patch *without* support for single platform builds, and tackle that as a subsequent PR - I'm guessing a mac-specific app option `universal2_build` which is True by default (for now), but can be configured to False, which reverts the install process to "old style" single platform installs; at some point in the future, we change the default, then deprecate the option.

However, if you think this is a feature we can't ship without including, I'll work up the modifications.

Reply:
I agree we can postpone this for now, but it would be worth creating an issue to track it.